v1.24.2: Preventing `ddev config` in parent directory has significant side-effects

### Expected Behavior

See Drupal Slack @kevinquillen : https://drupal.slack.com/archives/C5TQRQZRR/p1737819544903659

`ddev config` should work even if a module of a Drupal project has a .ddev in it (which often happens when using the ddev/ddev-drupal-contrib add-on)

### Actual Behavior

In v1.24.2, if you do a `ddev composer install` that happens to pull in a component (like a Drupal module) which had a .ddev directory, you can't `ddev config` the project. 

The example in the Slack thread was:
```
Failed to get project(s): 'ddev config' is not allowed in /home/kevinquillen/Projects/cci because a project exists in the subdirectory /home/kevinquillen/Projects/cci/docroot/modules/contrib/svg_image_field
Run 'ddev stop --unlist' for all projects in subdirectories of the current directory first to reenable 'ddev config'
```

This is an unexpected result of the wonderful
* https://github.com/ddev/ddev/pull/6852

### Steps To Reproduce

In a Drupal 11 project:
```
ddev composer require --dev drupal/svg_image_field:dev-2.3.x
```

Then `ddev config` the project.

### Anything else?

_No response_

Also, I have started getting this a lot, perhaps this is a new issue. DDEV commands seem delayed.

```
 ddev delete -Oy svg-image-field           
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xa67cf8]

goroutine 1 [running]:
github.com/ddev/ddev/pkg/ddevapp.GetActiveProjects()
        /home/runner/work/ddev/ddev/pkg/ddevapp/utils.go:51 +0x238
github.com/ddev/ddev/pkg/ddevapp.GetProjects(0x0)
        /home/runner/work/ddev/ddev/pkg/ddevapp/utils.go:371 +0x8e
github.com/ddev/ddev/cmd/ddev/cmd.getRequestedProjectsExtended({0xc000392480, 0x1, 0x0?}, 0x0, 0x1)
        /home/runner/work/ddev/ddev/cmd/ddev/cmd/utils.go:35 +0xe9
github.com/ddev/ddev/cmd/ddev/cmd.init.func66(0x133e820?, {0xc000392480, 0x1, 0x2})
        /home/runner/work/ddev/ddev/cmd/ddev/cmd/delete.go:39 +0xf3
github.com/spf13/cobra.(*Command).execute(0x133e820, {0xc000392440, 0x2, 0x2})
        /home/runner/work/ddev/ddev/vendor/github.com/spf13/cobra/command.go:989 +0xa91
github.com/spf13/cobra.(*Command).ExecuteC(0x133cb60)
        /home/runner/work/ddev/ddev/vendor/github.com/spf13/cobra/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/work/ddev/ddev/vendor/github.com/spf13/cobra/command.go:1041
github.com/ddev/ddev/cmd/ddev/cmd.Execute()
        /home/runner/work/ddev/ddev/cmd/ddev/cmd/root.go:122 +0x1a
main.main()
        /home/runner/work/ddev/ddev/cmd/ddev/main.go:27 +0xd3
```

I waited a few seconds, tried again, and it worked.

Same for Drush:

```
 [success] Successfully uninstalled: config_patch, config_patch_gitlab_api
➜  cci git:(feature/card-entity) ddev drush cex             
Error: unknown command "drush" for "ddev"
Run 'ddev --help' for usage.
➜  cci git:(feature/card-entity) ddev drush cex
Error: unknown command "drush" for "ddev"
Run 'ddev --help' for usage.
➜  cci git:(feature/card-entity) ddev drush cex
Error: unknown command "drush" for "ddev"
Run 'ddev --help' for usage.
```

Eventually, `ddev drush (cmd)` will work.

--- a/cmd/ddev/cmd/config_test.go
+++ b/cmd/ddev/cmd/config_test.go
@@ -441,25 +441,25 @@ func TestCmdConfigHasAllowedLocation(t *testing.T) {
 	// Make sure we're not allowed to config in home directory.
 	home, _ := os.UserHomeDir()
 	err = os.Chdir(home)
-	assert.NoError(err)
+	require.NoError(t, err)
 	out, err := exec.RunHostCommand(DdevBin, "config", "--project-type=php")
-	assert.Error(err)
-	assert.Contains(out, "not allowed in")
+	require.Error(t, err)
+	require.Contains(t, out, "not allowed in")
 
 	// Create a temporary directory and switch to it.
 	tmpDir := testcommon.CreateTmpDir(t.Name())
 	err = os.Chdir(tmpDir)
-	assert.NoError(err)
+	require.NoError(t, err)
 	// Create a project
 	_, err = exec.RunHostCommand(DdevBin, "config", "--project-type=php", "--project-name="+t.Name())
-	assert.NoError(err)
+	require.NoError(t, err)
 
 	t.Cleanup(func() {
 		err = os.Chdir(origDir)
 		assert.NoError(err)
 		_, err = exec.RunHostCommand(DdevBin, "delete", "-Oy", t.Name())
 		assert.NoError(err)
-		_, err = exec.RunHostCommand(DdevBin, "delete", "-Oy", t.Name()+"_subdir")
+		_, err = exec.RunHostCommand(DdevBin, "delete", "-Oy", t.Name()+"-subdir")
 		assert.NoError(err)
 		_ = os.RemoveAll(tmpDir)
 	})
@@ -467,25 +467,45 @@ func TestCmdConfigHasAllowedLocation(t *testing.T) {
 	subdirName := t.Name() + fileutil.RandomFilenameBase()
 	subdir := filepath.Join(tmpDir, subdirName)
 	err = os.Mkdir(subdir, 0777)
-	assert.NoError(err)
+	require.NoError(t, err)
 	err = os.Chdir(subdir)
-	assert.NoError(err)
+	require.NoError(t, err)
 
 	// Make sure that 'ddev config' in a subdir doesn't give an error for existing project
 	_, err = exec.RunHostCommand(DdevBin, "config", "--web-environment-add=FOO=BAR")
-	assert.NoError(err)
-	assert.NoFileExists(filepath.Join(subdir, ".ddev/config.yaml"))
-	assert.FileExists(filepath.Join(tmpDir, ".ddev/config.yaml"))
+	require.NoError(t, err)
+	require.NoFileExists(t, filepath.Join(subdir, ".ddev/config.yaml"))
+	require.FileExists(t, filepath.Join(tmpDir, ".ddev/config.yaml"))
 	configFileContents, err := os.ReadFile(filepath.Join(tmpDir, ".ddev/config.yaml"))
-	assert.NoError(err)
-	assert.Contains(string(configFileContents), "FOO=BAR")
+	require.NoError(t, err)
+	require.Contains(t, string(configFileContents), "FOO=BAR")
 
 	// Make sure that 'ddev config' in a subdir gives an error for new projects
-	out, err = exec.RunHostCommand(DdevBin, "config", "--project-type=php", "--project-name="+t.Name()+"_subdir")
-	assert.Error(err)
-	assert.Contains(out, fmt.Sprintf("project root '%s' already contains a project", tmpDir))
-	assert.Contains(out, "You may want to remove the existing project")
-	assert.NoFileExists(filepath.Join(subdir, ".ddev/config.yaml"))
+	out, err = exec.RunHostCommand(DdevBin, "config", "--project-type=php", "--project-name="+t.Name()+"-subdir")
+	require.Error(t, err)
+	require.Contains(t, out, fmt.Sprintf("project root '%s' already contains a project", tmpDir))
+	require.Contains(t, out, "You may want to remove the existing project")
+	require.NoFileExists(t, filepath.Join(subdir, ".ddev/config.yaml"))
+
+	// Create a dummy config file to make it look like it's an existing project
+	err = os.MkdirAll(filepath.Join(subdir, ".ddev"), 0777)
+	require.NoError(t, err)
+	_, err = os.Create(filepath.Join(subdir, ".ddev/config.yaml"))
+	require.NoError(t, err)
+
+	// Make sure that 'ddev config' in the subdir works fine for existing projects
+	out, err = exec.RunHostCommand(DdevBin, "config", "--project-type=php", "--project-name="+t.Name()+"-subdir")
+	require.NoError(t, err)
+	require.Contains(t, out, "'"+t.Name()+"-subdir'")
+	require.Contains(t, out, "Configuration complete")
+
+	// Make sure that 'ddev config' in the parent directory works fine for existing projects
+	err = os.Chdir(tmpDir)
+	require.NoError(t, err)
+	out, err = exec.RunHostCommand(DdevBin, "config", "--auto")
+	require.NoError(t, err)
+	require.Contains(t, out, "'"+t.Name()+"'")
+	require.Contains(t, out, "Configuration complete")
 }
 
 // TestConfigDatabaseVersion checks to make sure that both

--- a/pkg/ddevapp/config.go
+++ b/pkg/ddevapp/config.go
@@ -1418,6 +1418,10 @@ func HasAllowedLocation(app *DdevApp) error {
 		if app.AppRoot == project.AppRoot {
 			return nil
 		}
+		// If this is an existing project, allow it
+		if fileutil.FileExists(app.GetConfigPath("config.yaml")) {
+			return nil
+		}
 		// Do not allow 'ddev config' in any parent directory of any project
 		rel, err = filepath.Rel(app.AppRoot, project.AppRoot)
 		if err == nil && !strings.HasPrefix(rel, "..") {

--- a/pkg/ddevapp/config.go
+++ b/pkg/ddevapp/config.go
@@ -1402,6 +1402,11 @@ func HasAllowedLocation(app *DdevApp) error {
 	if fileutil.FileExists(filepath.Join(app.AppRoot, "cmd/ddev/main.go")) && fileutil.FileExists(filepath.Join(app.AppRoot, "cmd/ddev_gen_autocomplete/ddev_gen_autocomplete.go")) {
 		return fmt.Errorf("a project cannot be created in the DDEV source code (%v)", app.AppRoot)
 	}
+	// If this is an existing project, allow it.
+	if fileutil.FileExists(app.GetConfigPath("config.yaml")) {
+		return nil
+	}
+	// Check all projects if they are located in the subdirectories of the project we are in.
 	projectMap := globalconfig.GetGlobalProjectList()
 	projectList := make([]*globalconfig.ProjectInfo, 0, len(projectMap))
 	for _, project := range projectMap {
@@ -1416,10 +1421,6 @@ func HasAllowedLocation(app *DdevApp) error {
 		// Without sorting, a parent directory might be matched first,
 		// causing the function to return without checking the project in the subdirectory.
 		if app.AppRoot == project.AppRoot {
-			return nil
-		}
-		// If this is an existing project, allow it
-		if fileutil.FileExists(app.GetConfigPath("config.yaml")) {
 			return nil
 		}
 		// Do not allow 'ddev config' in any parent directory of any project

